test/jdk/sun/security/krb5/auto/SaslBasic.java

```diff
@@ -104,13 +100,68 @@ public void handle(Callback[] callbacks)
                     || lastClientToken[19] != 0) {
                 throw new Exception("maximum size for auth must be 0");
             }
+            testWrapUnwrapNoSecLayer(sc, ss);
         } else {
-            byte[] hello = "hello".getBytes();
-            token = sc.wrap(hello, 0, hello.length);
-            token = ss.unwrap(token, 0, token.length);
-            if (!Arrays.equals(hello, token)) {
-                throw new Exception("Message altered");
-            }
+            testWrapUnwrapWithSecLayer(sc, ss);
+        }
+    }
+
+    private static void testWrapUnwrapWithSecLayer(SaslClient sc, SaslServer ss) throws Exception {
+        byte[] token;
+        byte[] hello = "hello".getBytes();
+
+        // test client wrap and server unwrap
+        token = sc.wrap(hello, 0, hello.length);
+        token = ss.unwrap(token, 0, token.length);
+
+        if (!Arrays.equals(hello, token)) {
+            throw new Exception("Client message altered");
```

Comment:
please update these final exceptions from test to `RuntimeException`

Reply:
Updated to RuntimeException